symfit/core/minimizers.py

```diff
@@ -38,14 +39,15 @@ def execute(self, **options):
 
     @property
     def initial_guesses(self):
```

Comment:
Consider using `symfit.core.support.cache` to remove the boilerplate in checking for `_initial_guesses`. This way you also don't have to do `self._initial_guesses = None`. Adding the setter still works with this since you modify the `cache`.

Use as 
```python
@property
@cache
def initial_guesses(self):
    return [p.value for p in self.params]
```

Reply:
I think that would obfuscate the reason for doing this in the first place.

Reply:
Perhaps you are right. But what about a try except instead of if else then? Better to ask for forgiveness than permission. Saves the initiation and is more pythonic.

Reply:
I don't see where the try/except should go...

Reply:
```python
@property
def initial_guesses(self):
    if self._initial_guesses:
        return self._initial_guesses
    else:
        return [p.value for p in self.params]
```
should be
```python
@property
def initial_guesses(self):
    try:
        return self._initial_guesses
    except AttributeError:
        return [p.value for p in self.params]
```

Reply:
Jep, that's (more) elegant.